core/modules/file/file.field.inc

```diff
@@ -459,6 +475,7 @@ function file_field_widget_form(&$form, &$form_state, $field, $instance, $langco
       $elements[$delta]['#default_value'] = $defaults;
       $elements[$delta]['#weight'] = $delta;
       $elements[$delta]['#required'] = ($element['#required'] && $delta == 0);
+      $elements[$delta]['#multiple'] = TRUE;
```

Comment:
...again, unless I am missing something, according to https://api.backdropcms.org/form_api#multiple, `#multiple` applies only to `select`, and `tableselect`; not to files.